core/src/gas.rs

```diff
@@ -205,7 +205,7 @@ impl ValueCounter {
 }
 
 /// Gas allowance counter with some predefined maximum value.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
```

Comment:
```suggestion
#[derive(Clone, Debug)]
```
Why? Just requery it on-chain - it's free

Reply:
Yep, but this is required by `PrechargedDispatch` to be clonable to pass it further to `prepare` method.

Reply:
But we can query gas_allowance everytime to make sure that we hadn't somehow mutated it when it wasn't changed by accident

Reply:
Sorry seems I explained incorrectly =).

`precharge` method creates two counters: gas and allowance. These counters are charged for reading Program struct from storage and then go to `PrechargedDispatch` and then to `MessageExecutionContext` to be passed to `prepare` method. Between calls to `precharge` and `prepare` GasAllowance doesn't get updated on chain and frankly speaking it doesn't need to be updated since `prepare` method requires two precharged  counters.